File: go/circulo_struct/circulo.go
package main

import (
  "fmt"
  "math"
  "strconv"
  )
/*
*
* @author  Jorge Eduardo Ascencio Espíndola
* @version v1.0
*/

/*
* Declaración de la estrcutura Punto
*/
type Punto struct {
  x int // valor del eje X
  y int // valor del eje Y
}

/*
* Declaración de la estructura Circulo con centro en el plano XY
*/
type Circulo struct {
  centro Punto
  r float64
}

/*
* Cálcula el área del circulo c
*
* @param c : Circulo   circulo a calcular
*
* @return área del circulo
*/
func area(c Circulo) float64 {
  return math.Pi * c.r * c.r
}

func main(){
  var x_e string
  getX:
    for {
      fmt.Print("valor de x   ")
      fmt.Scanf("%s", &x_e)
      if isInt(x_e){
        break getX
      }
    }
  var y_e string
  getY:
    for {
      fmt.Print("valor de y   ")
      fmt.Scanf("%s", &y_e)
      if isInt(y_e){
	break getY
      }
    }
  x,_ := strconv.Atoi(x_e)
  y,_ := strconv.Atoi(y_e)
  p := Punto{x, y}
  var r_e string
  getR:
    for {
      fmt.Println("Valor del radio   ")
      fmt.Scanf("%s", &r_e)
      if isFloat(r_e){
        break getR
      }
    }
  r,_ := strconv.ParseFloat(r_e, 64)
  c := Circulo{p, r}
  fmt.Println(area(c))
}

/*
* Verifica si la cadena s es un entero
*/
func isInt(s string) bool {
  _, err := strconv.Atoi(s)
  return err == nil
}

/*
* Verifica si la cadena s es un flotante
*/
func isFloat(s string) bool {
  _, err := strconv.ParseFloat(s,64)
  return err == nil
}

File: julia/circulo_struct/circulo.jl
#!/usr/bin/julia

#=
 * Definición de la estructura Punto
=#
struct Punto
  x::Int32
  y::Int32
end

#=
 * Definición de la estructura Circulo
=#
struct Circulo
  centro::Punto
  radio::Float64
end

function area(c::Circulo)
  return pi * c.radio * c.radio
end

# Verifica que la entrada sea un entero
function isInt(str)
  return ismatch(r"^(-)?[0-9]+$", str) # verifica que la cadena sea valida
end

# Verifica que la entrada sea un flotante
function isFloat(str)
  return ismatch(r"^(-)?[0-9]+(.[0-9]*)?$", str) # verifica que la cadena sea valida
end

x = 0
while true
  print("valor de x:   ")
  x_e = readline(STDIN)
  if isInt(x_e)
    x = parse(Int, x_e)
    break
  end
end
y = 0
while true
  print("valor de y:   ")
  y_e = readline(STDIN)
  if isInt(y_e)
    y = parse(Int, y_e)
    break
  end
end
r = 0
while true
  print("valor del radio:   ")
  r_e = readline(STDIN)
  if isFloat(r_e)
    r = parse(Float64, r_e)
    break
  end
end
println(area(Circulo(Punto(x,y), r)))